### DISCUSSION TITLE
User Id on onRecordBeforeUpdateRequest hook

### DISCUSSION THREAD

**Discussion by revii1** [pocketbase/pocketbase#5356] [Q&A] [answered]
Hi,

I'm searching, and trying but I can't find any anwer. 

Is there any way to get authenticated `user id` in `onRecordBeforeUpdateRequest`??

**Comment by ganigeorgiev** [comment, marked as answer]
You can access the request data using the `e.httpContext` field. For retrieving the auth state you could check https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state (`c` in the examples is `e.httpContext`). For example:

```js
onRecordBeforeUpdateRequest((e) => {
    const info   = $apis.requestInfo(e.httpContext);
    const admin  = info.admin;      // empty if not authenticated as admin (note: with the next release admins are regular auth records)
    const record = info.authRecord; // empty if not authenticated as regular auth record

    console.log("Admin auth id:", admin?.id)
    console.log("Regular user auth id:", record?.id)
})
```

Note that the above is simplified as part of the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and in the upcoming v0.23.0 (_there are no ETAs yet_) it will look something like:

```js
onRecordUpdateRequest((e) => {
    const authRecord = e.auth()
    
    // to check whether it is "admin"/superuser you could use authRecord.isSuperuser()
    console.log("Auth id:", authRecord?.id)

    e.next()
})
```

**Reply by revii1** [reply]
Thank you very much, for fast response.
